pkg/process/process_manager.go

```diff
@@ -436,12 +437,20 @@ func (pm *Manager) ProcessReplace(ctx context.Context, req *rpc.ProcessReplaceRe
 		return nil, fmt.Errorf("Failed to start replacement process %v", p.Name)
 	}
 	pm.lock.Lock()
-
-	oldProcess := pm.processes[p.Name]
-	oldProcess.StopWithSignal(terminateSignal)
-	//TODO wait for the old process to stop
-	pm.releaseProcessPorts(oldProcess)
-	logrus.Infof("Process Manager: successfully unregistered old process %v", p.Name)
+	if processToReplace, exists := pm.processes[p.Name]; exists {
+		if existingProcess.UUID == processToReplace.UUID {
+			existingProcess.StopWithSignal(terminateSignal)
+			//TODO wait for the old process to stop
+			pm.releaseProcessPorts(existingProcess)
+			logrus.Infof("Process Manager: successfully unregistered old process %v", p.Name)
+		} else {
+			logrus.Warnf("Process Manager: replace process %v the existing process with UUID %v must have already been unregistered found process with UUID %v",
+				p.Name, existingProcess.UUID, processToReplace.UUID)
+		}
+	} else {
+		logrus.Warnf("Process Manager: replace process %v the existing process with UUID %v no longer exists",
+			p.Name, existingProcess.UUID)
+	}
```

Comment:
If this mismatching happens and there is no cleanup/rollback here, the existing process will become an orphan running process after the unregistering.